src/main/java/usersandpostsCRUD/demo/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,22 @@
+package usersandpostsCRUD.demo;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import usersandpostsCRUD.demo.exception.PostNotFoundException;
+import usersandpostsCRUD.demo.exception.UserNotFoundException;
+
+@ControllerAdvice
+public class GlobalExceptionHandler {
+
+    @ExceptionHandler(UserNotFoundException.class)
+    public ResponseEntity<String> handleUserNotFoundException(UserNotFoundException ex) {
+        return new ResponseEntity<>(ex.getMessage(), HttpStatus.NOT_FOUND);
```

Comment:
When something is wrong with the request, you return BAD REQUEST status code. Read about status code best practices